log/logrotate/logrotate.go

```diff
@@ -188,6 +198,13 @@ func (l *Logrotate) Stop() {
 	l.stop <- true
 }
 
+// AddSubscriber appends a subscriber to handle events.
+// The user can add multiple subscribers, each subcriber
+// will be called in the order as it is appended.
+func (l *Logrotate) AddSubscriber(sub Subscriber) {
```

Comment:
Since we went with the simple interface, we might as well make it even easier by ditching the interface in favour of a function type.

How about this?

```go
func (l *Logrotate) Register(fn func(string) error) {}
```